### PR TITLE
Added Description Property and DatabaseInterface Tests

### PR DESCRIPTION
**Pull request by mazin-s**
**What this PR does / why we need it**:

Adds optional description property and tests to DatabaseInterface methods.

**How Has This Been Tested?**:

1. Provisioned database with k8s. Here is the "kubectl get dbs" result
Screenshot 2023-07-24 at 1 29 43 PM

2. Database is provisioned as seen in era UI
Screenshot 2023-07-24 at 1 31 48 PM

3. Response from the post call using the nodejs app/
- I pulled the "best-app" form Manav's Docker Hub, created it, connected it to the database service that is created by the operator, and verified that the app is able to successfully read the database's contents.

test-proof
Screenshot 2023-07-24 at 1 37 16 PM

4. Webhook Test
image

### CODE REVIEW

File: config/samples/ndb_v1alpha1_database.yaml
Comment (shenoypritika) [inline config/samples/ndb_v1alpha1_database.yaml:1]: Some changes here:

- Please revert your local changes to the db name and sla name. 
- Please add the description field in the databaseInstance spec with a sample value.

Comment (shenoypritika) [reply config/samples/ndb_v1alpha1_database.yaml:1]: Also update the yaml in the READ.me with the new field

**Comment by shenoypritika** [comment]
- Please modify the yaml in the READMe
- Please describe your tests in a line or two and attach a screenshot of
      - provisioned db in k8s using `kubectl get dbs`
      - provisioned db in the era UI
      - Response from the post call using the nodejs app (similar to the one you have attached)
- Please fix the title of the PR

**Review by shenoypritika — CHANGES_REQUESTED** [review]

**Comment by mazin-s** [comment]
> * Please modify the yaml in the READMe
> * Please describe your tests in a line or two and attach a screenshot of
>   - provisioned db in k8s using `kubectl get dbs`
>   - provisioned db in the era UI
>   - Response from the post call using the nodejs app (similar to the one you have attached)
> * Please fix the title of the PR

Resolved

File: api/v1alpha1/database_types.go [hunk 1]
```diff
@@ -98,11 +101,10 @@ type Instance struct {
 	TimeZone string `json:"timezone"`
 	Type     string `json:"type"`
 	// +optional
-	Profiles Profiles `json:"profiles"`
+	Profiles *Profiles `json:"profiles"`
```
Comment (akshmish) [inline api/v1alpha1/database_types.go:104]: Why are these changes showing up in this PR? These don't seem to be related to this PR...

File: api/v1alpha1/database_webhook.go [hunk 2]
```diff
@@ -42,65 +44,69 @@ func (r *Database) SetupWebhookWithManager(mgr ctrl.Manager) error {
...
+func instanceSpecDefaulterForCreate(instance *Instance) {
 
-	if len(r.Spec.Instance.DatabaseNames) == 0 {
-		r.Spec.Instance.DatabaseNames = api.DefaultDatabaseNames
+	if len(instance.DatabaseNames) == 0 {
+		instance.DatabaseNames = api.DefaultDatabaseNames
 	}
 
-	if r.Spec.Instance.TimeZone == "" {
-		r.Spec.Instance.TimeZone = "UTC"
```
Comment (akshmish) [inline api/v1alpha1/database_webhook.go:52]: Same here, many of the Webhooks PR change are showing up here. Let's discuss it offline.

**Review by manavrajvanshi — APPROVED** [review]
Thanks for adding more unit tests.

**Review by shenoypritika — APPROVED** [review]

File: controller_adapters/database.go [hunk 3]
```diff
@@ -47,6 +47,15 @@ func (d *Database) GetDBInstanceName() string {
 	return d.Spec.Instance.DatabaseInstanceName
 }
 
+// Returns database description. If description is empty, creates a description
+func (d *Database) GetDBInstanceDescription() string {
+	description := d.Spec.Instance.Description
+	if description == "" {
+		return "Database provisioned by ndb-operator: " + d.GetDBInstanceName()
```
Comment (akshmish) [inline controller_adapters/database.go:54]: Should we also add this in the database_webhook instance defaulter method?

**Pull request merged**